### PR TITLE
chore: introduce bundle optimisations

### PR DESCRIPTION
**Pull request by wardpeet**
Introducing microbundle to have a simple bundler to create a gatsby-interface package.

- [x] removed `import { from } 'rxjs'`
- [x] introduce microbundle bundler`

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,5 +1,3 @@
-import { from } from "rxjs"
```
Comment (DSchau) [inline src/index.js]: Eek -- was this being included in the final bundle? Good fix nonetheless!

Comment (wardpeet) [reply src/index.js]: yes

**Comment by wardpeet** [comment]
I broke tests, i think but I don't really know what's going on here 😛 

image

help @rizchelwood

**Comment by Elanhant** [comment]
@wardpeet we've merged TypeScript into master, and I've tried to update this PR to work with TS but there were some issues that I was unable to resolve. I'm getting
> (babel plugin) Error: [BABEL] /Users/aleksei/work/gatsby-interface/src/index.ts: .value is not a valid Plugin property

when I run `yarn build`.

Let me know if you have time to discuss how to make this new build process work with TypeScript 🙏

My attempts can be found in [chore/bundle-optimisation-ts](https://github.com/gatsby-inc/gatsby-interface/compare/chore/bundle-optimisation-ts?expand=1) branch.

**Comment by mfrachet** [comment]
Hey there! Since it's opened for a long time, I'm wondering what is the status of this PR folks? 😊

**Comment by Elanhant** [comment]
Closing this since we've changed our bundling strategy in favour of preserving modules (#228) to avoid tree-shaking problems with Webpack.

**Pull request closed without merging**